File: packages/sdks/python/tester-local/tests/test_create_document_reference.py
import os

from generated.client import Metriport
from generated.resources.fhir import DocumentReference, CodeableConcept, Coding, DocumentReferenceContext, DocumentReferenceContent, Attachment

import os
from dotenv import load_dotenv

load_dotenv()

api_key = os.environ.get("API_KEY")
patient_id = os.environ.get("PATIENT_ID")
facility_id = os.environ.get("FACILITY_ID")
base_url = os.environ.get("BASE_URL")


def test_create_document_reference() -> None:
    metriport = Metriport(api_key=api_key, base_url=base_url)


    coding = Coding (
        system="http://loinc.org",
        code="3141-9",
        display="Body weight Measured"
    )

    # Create a CodeableConcept instance
    codeable_concept = CodeableConcept(
        coding=[coding],
    )

    facility_type = CodeableConcept (
        text="John Snow Clinic - Acute Care Centre"
    )

    document_reference_context = DocumentReferenceContext(
        period= {
            "start": "2023-10-10T14:14:17Z",
            "end": "2023-10-10T15:30:30Z"
        },
        facilityType=facility_type
    )

    document_reference_content = DocumentReferenceContent(attachment=Attachment())

    uploadDocumentReference = DocumentReference(
        resource_type="DocumentReference",
        content=[document_reference_content],
        description="Third degree wrist burn treatment",
        type=codeable_concept,
        context=document_reference_context
    )

    response = metriport.medical.document.create_document_reference(
        patient_id=patient_id,
        request=uploadDocumentReference
    )
    print(f"Response: {response}")

File: packages/sdks/java/tester-local/src/test/java/com.metriport/TestCreateDocumentReference.java
package com.metriport;

import com.metriport.generated.Metriport;
import com.metriport.generated.resources.fhir.types.Coding;
import com.metriport.generated.resources.fhir.types.CodeableConcept;
import com.metriport.generated.resources.medical.document.requests.UploadDocumentRequest;
import com.metriport.generated.resources.fhir.types.Period;
import com.metriport.generated.resources.fhir.types.Attachment;
import com.metriport.generated.resources.fhir.types.DocumentReferenceContent;
import com.metriport.generated.resources.fhir.types.DocumentReferenceContext;
import com.metriport.generated.resources.fhir.types.DocumentReference;


import org.junit.jupiter.api.Test;
import io.github.cdimascio.dotenv.Dotenv;

import java.util.Collections;

public class TestCreateDocumentReference {

    @Test
    public void test_start_doc_query() {
        Dotenv dotenv = Dotenv.load();
        Metriport client = Metriport.builder()
                .apiKey(dotenv.get("API_KEY"))
                .url(dotenv.get("BASE_URL"))
                .build();

        Coding coding = Coding.builder()
                .system("http://loinc.org")
                .code("3141-9")
                .display("Body weight Measured")
                .build();

        CodeableConcept typeCodeableConcept = CodeableConcept.builder()
                .coding(Collections.singletonList(coding))
                .build();

        CodeableConcept facilityType = CodeableConcept.builder()
                .text("John Snow Clinic - Acute Care Centre")
                .build();

        Period period = Period.builder()
                .start("2023-10-10T14:14:17Z")
                .end("2023-10-10T15:30:30Z")
                .build();

        DocumentReferenceContext documentReferenceContext = DocumentReferenceContext.builder()
                .period(period)
                .facilityType(facilityType)
                .build();

        Attachment attachment = Attachment.builder().build();

        DocumentReferenceContent documentReferenceContent = DocumentReferenceContent.builder()
                .attachment(attachment)
                .build();

        DocumentReference uploadDocumentReference = DocumentReference.builder()
                .resourceType("DocumentReference")
                .type(typeCodeableConcept)
                .context(Collections.singletonList(documentReferenceContext))
                .content(Collections.singletonList(documentReferenceContent))
                .build();

        UploadDocumentRequest uploadDocumentRequest = UploadDocumentRequest.builder()
                .patientId(dotenv.get("PATIENT_ID"))
                .body(uploadDocumentReference)
                .build();

        var response = client.medical().document().createDocumentReference(uploadDocumentRequest);

        System.out.println("Response: " + response);
    }
}